.github: update pull request template to add release note (#11848)
pingcap/tidb: .github/pull_request_template.md

 - Need to update the documentation
 - Need to update the `tidb-ansible` repository

>  - Need to be included in the release note

Reviewer: Remove it?

Author: done.

Release note

>  - Write release note if need cherry-pick or new feature.

Reviewer 2: ```suggestion
 - Write release note for bug-fix or new feature.
```
